src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterTypes } from './types/FilterTypes';
+
+function filterTodos(todos: Todo[], todosActivityFilter: FilterTypes, todosQuery: string) {
+  let resultTodos = [...todos];
+  const query = todosQuery.toLowerCase();
+
+  switch (todosActivityFilter) {
+    case 'completed':
+      resultTodos = resultTodos.filter(todo => todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    case 'active':
+      resultTodos = resultTodos.filter(todo => !todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    default:
+      resultTodos = resultTodos.filter(todo => todo.title.toLowerCase().includes(query));
+      break;
+  }
+
```

Comment:
I would filter with query after the `switch` statement so that you don't repet yourself